WordPress/gutenberg
# `DisclosureContent`: migrate from `reakit` to `@ariakit/react`

## What?

- Migrate `DisclosureContent` from `reakit` to `ariakit`
- Migrate to TypeScript

## Why?

Part of #53278

As stated in the linked issue [Reakit](https://reakit.io/) does not explicitly support React 18 (causing peer dependency warnings) and Reakit has already been succeeded by [Ariakit](https://ariakit.org/), so it will not receive any more updates.

## How?

Switch out the Reakit dependency for Ariakit. We also have to create a small wrapper to maintain the existing API of the component.

## Testing Instructions

This component doesn't come with a story and is used in only one place:

- Locally run `npm run dev` and `npm run wp-env start`
- Go to `localhost:8888` and activate a theme which supports widgets
- Go to `Appearance` > `Widgets` and verify toggling widget areas still works as expected
